docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/TracerV-with-FlameGraph.rst

```diff
@@ -0,0 +1,158 @@
+.. _tracerv-with-flamegraphs:
+
+TracerV + Flame Graphs: Profiling Software with Out-of-Band Flame Graph Generation
+====================================================================================
+
+FireSim supports generating `Flame Graphs
+<http://www.brendangregg.com/flamegraphs.html>`_ out-of-band, to visualize
+the performance of software running on simulated processors. This feature was
+introduced in our `FirePerf paper at ASPLOS 2020
+<https://sagark.org/assets/pubs/fireperf-asplos2020.pdf>`_ .
+
+Before proceeding, make sure you understand the :ref:`tracerv` section.
+
+What are Flame Graphs?
+-----------------------
+
+.. figure:: http://www.brendangregg.com/FlameGraphs/cpu-mysql-updated.svg
+    :align: center
+    :alt: Example Flame Graph
+
+    Example Flame Graph (from http://www.brendangregg.com/FlameGraphs/)
+
+
+Flame Graphs are a type of histogram that shows where software is spending its
```

Comment:
I know it's not good marketing but i think the userspace limitations should be made more obvious.